frontend/components/Leaderboard.tsx

```diff
@@ -0,0 +1,194 @@
+import React, { useContext, useEffect, useState } from "react";
+import {
+  Box,
+  Avatar,
+  Flex,
+  Text,
+  Grid,
+  useBreakpointValue,
+  Skeleton,
+} from "@chakra-ui/react";
+import { inter } from "@/app/fonts"; // Make sure this import path is correct
+import { BurntPixArchives__factory } from "@/contracts";
+import { WalletContext } from "./wallet/WalletContext";
+import { getProfileData } from "@/utils/universalProfile";
+import { constants } from "@/constants/constants";
+
+interface Contribution {
+  contributor: string;
+  contribution: number;
+  upName: string | null;
+  avatar: string | null;
+}
+
+const Leaderboard: React.FC = () => {
+  const margin = useBreakpointValue({ base: "0", md: "20px" });
+  const walletContext = useContext(WalletContext);
+  const { networkConfig, provider } = walletContext;
+  const [sortedContributions, setSortedContributions] = useState<
+    Contribution[]
+  >([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const truncateName = (name: string) => {
+    if (name.length > 10) {
+      return name.substring(0, 10) + "...";
+    }
+    return name;
+  };
+
+  const burntPixArchives = BurntPixArchives__factory.connect(
+    networkConfig.burntPixArchivesAddress,
+    provider,
+  );
+
+  useEffect(() => {
+    const fetchAllContributions = async () => {
+      try {
+        const contributors = await burntPixArchives.getContributors();
+        console.log("Contributors", contributors);
+
+        const rpcUrl = networkConfig.rpcUrl;
+        const contributionsPromises = contributors.map((contributor) =>
+          fetchContributionAndProfile(contributor, rpcUrl),
```

Comment:
@Ledespana this is really inefficient in terms of the amount of calls that are happening to the RPC. if there are 10 contributors this would result in 20 calls (iterations + profile data being a call each). Instead we should be fetching all the contributions separately using BurntPicArchives.getContributions([address_1, address_2, etc.])